error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:10:28
   |
10 |     impl Unpin for A where PhantomPinned: Unpin {} //~ ERROR E0277
   |                            ^^^^^^^^^^^^^^^^^^^^ the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:18:28
   |
18 |     impl Unpin for B where Wrapper<PhantomPinned>: Unpin {} //~ ERROR E0277
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Unpin` is not implemented for `PhantomPinned`, which is required by `phantom_pinned::Wrapper<PhantomPinned>: Unpin`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required for `phantom_pinned::Wrapper<PhantomPinned>` to implement `Unpin`
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:14:13
   |
14 |     impl<T> Unpin for Wrapper<T> where T: Unpin {}
   |             ^^^^^     ^^^^^^^^^^          ----- unsatisfied trait bound introduced here
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:36:28
   |
36 |     impl Unpin for A where Inner: Unpin {} //~ ERROR E0277
   |                            ^^^^^^^^^^^^ within `Inner`, the trait `Unpin` is not implemented for `PhantomPinned`, which is required by `Inner: Unpin`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `Inner`
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:32:12
   |
32 |     struct Inner(PhantomPinned);
   |            ^^^^^
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:44:28
   |
44 |     impl Unpin for B where Wrapper<Inner>: Unpin {} //~ ERROR E0277
   |                            ^^^^^^^^^^^^^^^^^^^^^ within `Inner`, the trait `Unpin` is not implemented for `PhantomPinned`, which is required by `inner::Wrapper<Inner>: Unpin`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `Inner`
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:32:12
   |
32 |     struct Inner(PhantomPinned);
   |            ^^^^^
note: required for `inner::Wrapper<Inner>` to implement `Unpin`
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:40:13
   |
40 |     impl<T> Unpin for Wrapper<T> where T: Unpin {}
   |             ^^^^^     ^^^^^^^^^^          ----- unsatisfied trait bound introduced here
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
