error: #[repr(packed)] attribute should not be name-value pair
  --> tests/ui/pin_project/packed-name-value.rs:18:8
   |
18 | #[repr(packed = "")] //~ ERROR attribute should not be name-value pair
   |        ^^^^^^^^^^^

error: #[repr(packed)] attribute should not be name-value pair
  --> tests/ui/pin_project/packed-name-value.rs:23:8
   |
23 | #[repr(packed = "")] //~ ERROR attribute should not be name-value pair
   |        ^^^^^^^^^^^

error[E0693]: incorrect `repr(packed)` attribute format
  --> tests/ui/pin_project/packed-name-value.rs:18:8
   |
18 | #[repr(packed = "")] //~ ERROR attribute should not be name-value pair
   |        ^^^^^^^^^^^ help: use parentheses instead: `packed()`

error[E0693]: incorrect `repr(packed)` attribute format
  --> tests/ui/pin_project/packed-name-value.rs:23:8
   |
23 | #[repr(packed = "")] //~ ERROR attribute should not be name-value pair
   |        ^^^^^^^^^^^ help: use parentheses instead: `packed()`
