error: #[pin_project] attribute has been removed
  --> tests/ui/pin_project/remove-attr-from-struct.rs:24:1
   |
24 | #[pin_project] //~ ERROR has been removed
   | ^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot find attribute `pin` in this scope
  --> tests/ui/pin_project/remove-attr-from-struct.rs:20:7
   |
20 |     #[pin] //~ ERROR cannot find attribute `pin` in this scope
   |       ^^^

error: cannot find attribute `pin` in this scope
  --> tests/ui/pin_project/remove-attr-from-struct.rs:13:7
   |
13 |     #[pin] //~ ERROR cannot find attribute `pin` in this scope
   |       ^^^

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/remove-attr-from-struct.rs:37:16
   |
37 |     is_unpin::<A>(); //~ ERROR E0277
   |                ^ within `A`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `A`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:12:8
   |
12 | struct A {
   |        ^
note: required by a bound in `is_unpin`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:8:16
   |
8  | fn is_unpin<T: Unpin>() {}
   |                ^^^^^ required by this bound in `is_unpin`

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/remove-attr-from-struct.rs:38:16
   |
38 |     is_unpin::<B>(); //~ ERROR E0277
   |                ^ within `B`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `B`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:19:8
   |
19 | struct B {
   |        ^
note: required by a bound in `is_unpin`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:8:16
   |
8  | fn is_unpin<T: Unpin>() {}
   |                ^^^^^ required by this bound in `is_unpin`

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/remove-attr-from-struct.rs:42:22
   |
42 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |             -------- ^^^^^^ within `A`, the trait `Unpin` is not implemented for `PhantomPinned`
   |             |
   |             required by a bound introduced by this call
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `A`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:12:8
   |
12 | struct A {
   |        ^
note: required by a bound in `Pin::<Ptr>::new`
  --> $RUST/core/src/pin.rs
   |
   | impl<Ptr: Deref<Target: Unpin>> Pin<Ptr> {
   |                         ^^^^^ required by this bound in `Pin::<Ptr>::new`
...
   |     pub const fn new(pointer: Ptr) -> Pin<Ptr> {
   |                  --- required by a bound in this associated function

error[E0599]: no method named `project` found for struct `Pin<&mut A>` in the current scope
  --> tests/ui/pin_project/remove-attr-from-struct.rs:42:30
   |
42 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |                              ^^^^^^^ method not found in `Pin<&mut A>`

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/remove-attr-from-struct.rs:45:22
   |
45 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |             -------- ^^^^^^ within `B`, the trait `Unpin` is not implemented for `PhantomPinned`
   |             |
   |             required by a bound introduced by this call
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `B`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:19:8
   |
19 | struct B {
   |        ^
note: required by a bound in `Pin::<Ptr>::new`
  --> $RUST/core/src/pin.rs
   |
   | impl<Ptr: Deref<Target: Unpin>> Pin<Ptr> {
   |                         ^^^^^ required by this bound in `Pin::<Ptr>::new`
...
   |     pub const fn new(pointer: Ptr) -> Pin<Ptr> {
   |                  --- required by a bound in this associated function

error[E0599]: no method named `project` found for struct `Pin<&mut B>` in the current scope
  --> tests/ui/pin_project/remove-attr-from-struct.rs:45:30
   |
45 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |                              ^^^^^^^ method not found in `Pin<&mut B>`
