error[E0119]: conflicting implementations of trait `UnsafeUnpin` for type `Foo<_, _>`
  --> tests/ui/not_unpin/impl-unsafe-unpin.rs:5:1
   |
5  | #[pin_project(!Unpin)] //~ ERROR E0119
   | ^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Foo<_, _>`
...
12 | unsafe impl<T, U> UnsafeUnpin for Foo<T, U> where T: Unpin {}
   | ---------------------------------------------------------- first implementation here
   |
   = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `UnsafeUnpin` for type `Bar<_, _>`
  --> tests/ui/not_unpin/impl-unsafe-unpin.rs:14:1
   |
14 | #[pin_project(!Unpin)] //~ ERROR E0119
   | ^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Bar<_, _>`
...
21 | unsafe impl<T, U> UnsafeUnpin for Bar<T, U> {}
   | ------------------------------------------- first implementation here
   |
   = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `UnsafeUnpin` for type `Baz<_, _>`
  --> tests/ui/not_unpin/impl-unsafe-unpin.rs:23:1
   |
23 | #[pin_project(!Unpin)] //~ ERROR E0119
   | ^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Baz<_, _>`
...
30 | unsafe impl<T: Unpin, U: Unpin> UnsafeUnpin for Baz<T, U> {}
   | --------------------------------------------------------- first implementation here
   |
   = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)
