error[E0119]: conflicting implementations of trait `FooMustNotImplDrop` for type `Foo<_, _>`
 --> tests/ui/pin_project/conflict-drop.rs:7:1
  |
7 | #[pin_project] //~ ERROR E0119
  | ^^^^^^^^^^^^^^
  | |
  | first implementation here
  | conflicting implementation for `Foo<_, _>`
  |
  = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `Drop` for type `Bar<_, _>`
  --> tests/ui/pin_project/conflict-drop.rs:18:15
   |
18 | #[pin_project(PinnedDrop)] //~ ERROR E0119
   |               ^^^^^^^^^^ conflicting implementation for `Bar<_, _>`
...
30 | impl<T, U> Drop for Bar<T, U> {
   | ----------------------------- first implementation here
