error[E0119]: conflicting implementations of trait `Unpin` for type `Foo<_, _>`
  --> tests/ui/pin_project/conflict-unpin.rs:7:1
   |
7  | #[pin_project] //~ ERROR E0119
   | ^^^^^^^^^^^^^^ conflicting implementation for `Foo<_, _>`
...
15 | impl<T, U> Unpin for Foo<T, U> where T: Unpin {} // Conditional Unpin impl
   | --------------------------------------------- first implementation here
   |
   = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `Unpin` for type `Bar<_, _>`
  --> tests/ui/pin_project/conflict-unpin.rs:19:1
   |
19 | #[pin_project] //~ ERROR E0119
   | ^^^^^^^^^^^^^^ conflicting implementation for `Bar<_, _>`
...
27 | impl<T, U> Unpin for Bar<T, U> {} // Non-conditional Unpin impl
   | ------------------------------ first implementation here
   |
   = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `Unpin` for type `Baz<_, _>`
  --> tests/ui/pin_project/conflict-unpin.rs:29:1
   |
29 | #[pin_project] //~ ERROR E0119
   | ^^^^^^^^^^^^^^ conflicting implementation for `Baz<_, _>`
...
37 | impl<T: Unpin, U: Unpin> Unpin for Baz<T, U> {} // Conditional Unpin impl
   | -------------------------------------------- first implementation here
   |
   = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)
