error: expected item after attributes
 --> tests/ui/pin_project/packed_sneaky-3.rs:6:1
  |
6 | hidden_repr_macro! {} //~ ERROR expected item after attributes
  | ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `hidden_repr_macro` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected item after attributes
  --> tests/ui/pin_project/packed_sneaky-3.rs:15:9
   |
15 |         #[repr(packed)] //~ ERROR expected item after attributes
   |         ^^^^^^^^^^^^^^^
...
19 | hidden_repr_macro2! {}
   | ---------------------- in this macro invocation
   |
   = note: this error originates in the macro `hidden_repr_macro2` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected item after attributes
  --> tests/ui/pin_project/packed_sneaky-3.rs:26:10
   |
26 | #[derive(HiddenRepr)] //~ ERROR expected item after attributes
   |          ^^^^^^^^^^
   |
   = note: this error originates in the derive macro `HiddenRepr` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc-macro derive produced unparsable tokens
  --> tests/ui/pin_project/packed_sneaky-3.rs:26:10
   |
26 | #[derive(HiddenRepr)] //~ ERROR expected item after attributes
   |          ^^^^^^^^^^
