error: #[pin_project] attribute may not be used on #[repr(packed)] types
 --> tests/ui/pin_project/packed.rs:6:8
  |
6 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] types
  |        ^^^^^^

error: #[pin_project] attribute may not be used on #[repr(packed)] types
  --> tests/ui/pin_project/packed.rs:13:8
   |
13 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] types
   |        ^^^^^^

error: #[pin_project] attribute may not be used on #[repr(packed)] types
  --> tests/ui/pin_project/packed.rs:21:8
   |
21 | #[repr(packed(2))] //~ ERROR may not be used on #[repr(packed)] types
   |        ^^^^^^^^^

error: #[pin_project] attribute may not be used on #[repr(packed)] types
  --> tests/ui/pin_project/packed.rs:28:8
   |
28 | #[repr(packed(2))] //~ ERROR may not be used on #[repr(packed)] types
   |        ^^^^^^^^^
