error: trait bound PhantomPinned: Unpin does not depend on any type or lifetime parameters
  --> tests/ui/unstable-features/trivial_bounds.rs:18:43
   |
18 |     impl Unpin for A where PhantomPinned: Unpin {} //~ ERROR Unpin does not depend on any type or lifetime parameters
   |                                           ^^^^^
   |
note: the lint level is defined here
  --> tests/ui/unstable-features/trivial_bounds.rs:8:9
   |
8  | #![deny(trivial_bounds)]
   |         ^^^^^^^^^^^^^^

error: trait bound Inner: Unpin does not depend on any type or lifetime parameters
  --> tests/ui/unstable-features/trivial_bounds.rs:22:35
   |
22 |     impl Unpin for B where Inner: Unpin {} //~ ERROR Unpin does not depend on any type or lifetime parameters
   |                                   ^^^^^

error: trait bound Wrapper<Inner>: Unpin does not depend on any type or lifetime parameters
  --> tests/ui/unstable-features/trivial_bounds.rs:30:44
   |
30 |     impl Unpin for C where Wrapper<Inner>: Unpin {} //~ ERROR Unpin does not depend on any type or lifetime parameters
   |                                            ^^^^^
